app/lib/basecamp.rb

```diff
@@ -33,21 +33,33 @@ def resources
 
   def apply_todo_workflow(resource, pull_request)
     current_user_comments = comments_by_current_user(resource)
+    tags = pull_request.tags + ["Status:#{pull_request.state}", info_colour(pull_request.state)]
+
+    # Prevent multiple comments without state change
+    if pull_request.status == PullRequestStatus::UPDATED && current_user_comments.any? { |c| match_tags(c, tags) }
+      @logger.info "Comment already created on #{resource.type}##{resource.id}"
+      return
+    end
+
     case pull_request.status
-    when PullRequestStatus::OPENED, PullRequestStatus::UPDATED
```

Comment:
The `message` variable is not being created for the case when the PR status is UPDATED and this the first time the comment is going to be added